From ad93824d44b370316d021b1fadacffa56d85e8c3 Mon Sep 17 00:00:00 2001
From: James <aslan.jnn@gmail.com>
Date: Fri, 28 Apr 2023 22:10:20 +1000
Subject: Do not trim file name when opening simple open dialog

---
 .../services/dialogs/browser/abstractFileDialogService.ts       | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts b/src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts
index bb082edfd8d..87cf7b07607 100644
--- a/src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts
+++ b/src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts
@@ -68,9 +68,6 @@ export abstract class AbstractFileDialogService implements IFileDialogService {
 			if (candidate) {
 				this.logService.debug(`[FileDialogService] Default file path using last active workspace root: ${candidate}`);
 			}
-		} else {
-			this.logService.debug(`[FileDialogService] Default file path using parent of last active file: ${candidate}`);
-			candidate = resources.dirname(candidate);
 		}
 
 		if (!candidate) {
--
2.37.1 (Apple Git-137.1)
