From 8683c139deddf0b0b6c64502bdf1805c2d0cf30f Mon Sep 17 00:00:00 2001
From: James <aslan.jnn@gmail.com>
Date: Tue, 27 Jun 2023 13:51:52 +1000
Subject: Add option to enable Metal backend

---
 src/main.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/main.js b/src/main.js
index 5b3bcf8b9ab..c579a020ac4 100644
--- a/src/main.js
+++ b/src/main.js
@@ -39,6 +39,11 @@ if (args['sandbox']) {
 	app.enableSandbox();
 }
 
+// Enable Metal renderer backend via `--metal` argument
+if (args['metal']) {
+	app.commandLine.appendSwitch('use-angle', 'metal');
+}
+
 // Set userData path before app 'ready' event
 const userDataPath = getUserDataPath(args, product.nameShort ?? 'code-oss-dev');
 if (process.platform === 'win32') {
-- 
2.39.2 (Apple Git-144)
