From 17531bc6dcdc446b71a956cbbb92fb1181355ce0 Mon Sep 17 00:00:00 2001
From: James <aslan.jnn@gmail.com>
Date: Sun, 7 Jan 2024 17:21:18 +1100
Subject: Add toggle to mangle build phase

diff --git a/build/gulpfile.reh.js b/build/gulpfile.reh.js
index c1d64c00338..86afd47db16 100644
--- a/build/gulpfile.reh.js
+++ b/build/gulpfile.reh.js
@@ -26,7 +26,7 @@ const gunzip = require('gulp-gunzip');
 const File = require('vinyl');
 const fs = require('fs');
 const glob = require('glob');
-const { compileBuildWithManglingTask } = require('./gulpfile.compile');
+const { compileBuildWithManglingTask, compileBuildWithoutManglingTask } = require('./gulpfile.compile');
 const { cleanExtensionsBuildTask, compileNonNativeExtensionsBuildTask, compileNativeExtensionsBuildTask, compileExtensionMediaBuildTask } = require('./gulpfile.extensions');
 const { vscodeWebResourceIncludes, createVSCodeWebFileContentMapper } = require('./gulpfile.vscode.web');
 const cp = require('child_process');
@@ -473,7 +473,7 @@ function tweakProductForServerWeb(product) {
 			gulp.task(serverTaskCI);
 
 			const serverTask = task.define(`vscode-${type}${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
-				compileBuildWithManglingTask,
+				minified ? compileBuildWithManglingTask : compileBuildWithoutManglingTask,
 				cleanExtensionsBuildTask,
 				compileNonNativeExtensionsBuildTask,
 				compileExtensionMediaBuildTask,
diff --git a/extensions/mangle-loader.js b/extensions/mangle-loader.js
index b6b22ce3f1a..3de6721988e 100644
--- a/extensions/mangle-loader.js
+++ b/extensions/mangle-loader.js
@@ -37,7 +37,7 @@ function getMangledFileContents(projectPath) {
  * @type {webpack.LoaderDefinitionFunction}
  */
 module.exports = async function (source, sourceMap, meta) {
-	if (this.mode !== 'production') {
+	if (this.mode !== 'production' || !process.env['VSCODE_MINIFY']) {
 		// Only enable mangling in production builds
 		return source;
 	}
-- 
2.39.3 (Apple Git-145)
