From ac9e0f42088aaeff4d5ead51ce7fd4259e5f107e Mon Sep 17 00:00:00 2001
From: James <aslan.jnn@gmail.com>
Date: Sun, 13 Aug 2023 17:27:12 +1000
Subject: Enable raw draw feature

---
 src/main.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main.js b/src/main.js
index 825dbc5cb20..2ecc5a5c5a6 100644
--- a/src/main.js
+++ b/src/main.js
@@ -57,6 +57,7 @@ if (args['sandbox'] &&
 if (args['metal']) {
 	app.commandLine.appendSwitch('use-angle', 'metal');
 }
+app.commandLine.appendSwitch('enable-features', 'RawDraw');
 
 // Set userData path before app 'ready' event
 const userDataPath = getUserDataPath(args, product.nameShort ?? 'code-oss-dev');
-- 
2.39.2 (Apple Git-144)
