From 5136b1c41edd3d2fae9f6159ba3ff3f72ac5cec1 Mon Sep 17 00:00:00 2001
From: James <aslan.jnn@gmail.com>
Date: Sat, 1 Feb 2025 21:15:35 +1100
Subject: No reload on workspace change

---
 .../electron-sandbox/workspaceEditingService.ts | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/src/vs/workbench/services/workspaces/electron-sandbox/workspaceEditingService.ts b/src/vs/workbench/services/workspaces/electron-sandbox/workspaceEditingService.ts
index 625f69f60b9..9d996bcb3e5 100644
--- a/src/vs/workbench/services/workspaces/electron-sandbox/workspaceEditingService.ts
+++ b/src/vs/workbench/services/workspaces/electron-sandbox/workspaceEditingService.ts
@@ -43,6 +43,7 @@ export class NativeWorkspaceEditingService extends AbstractWorkspaceEditingServi
 		@INativeHostService private nativeHostService: INativeHostService,
 		@IWorkbenchConfigurationService configurationService: IWorkbenchConfigurationService,
 		@IStorageService private storageService: IStorageService,
+		// @ts-ignore
 		@IExtensionService private extensionService: IExtensionService,
 		@IWorkingCopyBackupService private workingCopyBackupService: IWorkingCopyBackupService,
 		@INotificationService notificationService: INotificationService,
@@ -174,11 +175,6 @@ export class NativeWorkspaceEditingService extends AbstractWorkspaceEditingServi
 	}
 
 	async enterWorkspace(workspaceUri: URI): Promise<void> {
-		const stopped = await this.extensionService.stopExtensionHosts(localize('restartExtensionHost.reason', "Opening a multi-root workspace."));
-		if (!stopped) {
-			return;
-		}
-
 		const result = await this.doEnterWorkspace(workspaceUri);
 		if (result) {
 
@@ -191,17 +187,6 @@ export class NativeWorkspaceEditingService extends AbstractWorkspaceEditingServi
 				this.workingCopyBackupService.reinitialize(newBackupWorkspaceHome);
 			}
 		}
-
-		// TODO@aeschli: workaround until restarting works
-		if (this.environmentService.remoteAuthority) {
-			this.hostService.reload();
-		}
-
-		// Restart the extension host: entering a workspace means a new location for
-		// storage and potentially a change in the workspace.rootPath property.
-		else {
-			this.extensionService.startExtensionHosts();
-		}
 	}
 }
 
-- 
2.39.5 (Apple Git-154)
