From 1e04f52d534ec6d648fe9ac6463d347208571cd4 Mon Sep 17 00:00:00 2001
From: James <aslan.jnn@gmail.com>
Date: Sat, 15 Feb 2025 09:04:15 +1100
Subject: Fix reload on workspace change

---
 .../workspaceEditingService.ts                 | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/src/vs/workbench/services/workspaces/electron-browser/workspaceEditingService.ts b/src/vs/workbench/services/workspaces/electron-browser/workspaceEditingService.ts
index 625f69f60b9..eeebceea967 100644
--- a/src/vs/workbench/services/workspaces/electron-browser/workspaceEditingService.ts
+++ b/src/vs/workbench/services/workspaces/electron-browser/workspaceEditingService.ts
@@ -174,11 +174,6 @@ export class NativeWorkspaceEditingService extends AbstractWorkspaceEditingServi
 	}
 
 	async enterWorkspace(workspaceUri: URI): Promise<void> {
-		const stopped = await this.extensionService.stopExtensionHosts(localize('restartExtensionHost.reason', "Opening a multi-root workspace"));
-		if (!stopped) {
-			return;
-		}
-
 		const result = await this.doEnterWorkspace(workspaceUri);
 		if (result) {
 
@@ -195,12 +190,13 @@ export class NativeWorkspaceEditingService extends AbstractWorkspaceEditingServi
 		// TODO@aeschli: workaround until restarting works
 		if (this.environmentService.remoteAuthority) {
 			this.hostService.reload();
-		}
-
-		// Restart the extension host: entering a workspace means a new location for
-		// storage and potentially a change in the workspace.rootPath property.
-		else {
-			this.extensionService.startExtensionHosts();
+		} else {
+			const stopped = await this.extensionService.stopExtensionHosts(localize('restartExtensionHost.reason', "Opening a multi-root workspace."));
+			if (stopped) {
+				// Restart the extension host: entering a workspace means a new location for
+				// storage and potentially a change in the workspace.rootPath property.
+				this.extensionService.startExtensionHosts();
+			}
 		}
 	}
 }
-- 
2.39.5 (Apple Git-154)
